### PR TITLE
scalars: use `tf-downloader` for download links

### LINKED ISSUE #1845: Scalar plugin download links should work in Colab
Currently, the scalar plugin download links use `<a href download>`,
where the `href` points to a route on the TensorBoard backend. This
works fine locally, but on Colab requests to the backend are mediated by
a service worker, and the service worker does not appear to intercept
these requests. Consequently, downloads fail with a “network error”.

Proposed fix (to apply to Colab contexts only): When a run is selected
from the “run to download” dropdown menu, instead of updating the `href`
property of the links, send off two asynchronous `fetch` requests: one
to the CSV data and one to the JSON data. As these resolve, update the
`href`s to point to data URLs/blob URLs.

### PR DESCRIPTION
**Pull request by wchargin**
Summary:
We have a `<tf-downloader>` element that provides CSV and JSON download
links alongside a run selector, but this element is actually unused.
This commit replaces the scalar plugin’s reimplementation, updating the
styling of `<tf-downloader>` to stay mostly consistent with the existing
UI. This will enable us to fix #1845 in a way that can work for other
download links, too.

Test Plan:
Observe that the UI is virtually identical (the only visible change is
that the vertical position of the dropdown menu has shifted slightly):

“Before” and “after” screenshot.

Check that the download links download identical files before and after
the change, with the same filenames except that the `run_` prefix has
been changed to `run-` for internal consistency.

wchargin-branch: scalars-tf-downloader

### CODE REVIEW

File: tensorboard/components/tf_dashboard_common/tf-downloader.html [hunk 1]
```diff
@@ -89,10 +81,10 @@
         return urlFn(tag, run);
       },
       _csvName(tag, run) {
-        return `run_${run},tag_${tag}.csv`;
+        return `run_${run}-tag-${tag}.csv`;
```
Comment (nfelt) [inline tensorboard/components/tf_dashboard_common/tf-downloader.html]: this uses an underscore for run but dashes for the rest - make it consistent?  ditto below

File: tensorboard/components/tf_dashboard_common/tf-downloader.html [hunk 2]
```diff
@@ -80,7 +72,7 @@
         _run: String,
         runs: Array,
         tag: String,
-        urlFn: Function,
+        urlFn: Function,  // (tag: string, run: string) => string (JSON URL)
```
Comment (nfelt) [inline tensorboard/components/tf_dashboard_common/tf-downloader.html]: Maybe move this comment up one line and just slightly expand to state that the user should provide this to generate URLs for the selected run/tag combos.

File: tensorboard/components/tf_dashboard_common/tf-downloader.html (hunk 1 above)
Comment (wchargin) [reply tensorboard/components/tf_dashboard_common/tf-downloader.html]: This was for consistency with the original scalar-card code

<https://github.com/tensorflow/tensorboard/blob/fc61089cefca2c909a5cf7de0fc3ac44e0320720/tensorboard/plugins/scalar/tf_scalar_dashboard/tf-scalar-card.html#L112>

but I’m fine with disregarding that.

File: tensorboard/components/tf_dashboard_common/tf-downloader.html (hunk 2 above)
Comment (wchargin) [reply tensorboard/components/tf_dashboard_common/tf-downloader.html]: Sure.

File: tensorboard/components/tf_dashboard_common/tf-downloader.html (hunk 1 above)
Comment (nfelt) [reply tensorboard/components/tf_dashboard_common/tf-downloader.html]: Ah okay.  I think we can just disregard that; I doubt anyone is depending on the exact format of that downloaded file name.

**Pull request merged**